src/modules/hip/kernel/flip.hpp

```diff
@@ -37,7 +38,13 @@ __global__ void flip_pkd_tensor(T *srcPtr,
     }
     else if(horizontalFlag == 1)
     {
-        srcIdx += ((id_y + roiTensorPtrSrc[id_z].ltrbROI.lt.y) * srcStridesNH.y) + (roiTensorPtrSrc[id_z].ltrbROI.rb.x - id_x - 7) * 3;
+        if((id_z == 0) && (id_y == 0) && (id_x + 8) > roiTensorPtrSrc[id_z].xywhROI.roiWidth)
```

Comment:
Ensure proof and reasoning for minimal performance reduction on addition of this if